### PR TITLE
[Refactor] stream decompression

### LINKED ISSUE #8048: [Refactor] move decompressor to stream compression

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8048

## Problem Summary(Required) ：

1. move decompressor to stream decompress
2. add decompression pool for LZ4F and ZSTD

### CODE REVIEW

File: be/src/util/compression/stream_compression.h [hunk 1]
```diff
@@ -63,85 +69,94 @@ class Decompressor {
...
     virtual Status init() { return Status::OK(); }
 
-    Decompressor(CompressionTypePB ctype) : _ctype(ctype) {}
+    StreamCompression(CompressionTypePB ctype) : _ctype(ctype) {}
 
     CompressionTypePB _ctype;
 };
 
-class GzipDecompressor : public Decompressor {
+class GzipStreamCompression : public StreamCompression {
```
Comment (imay) [inline be/src/util/compression/stream_compression.h:77]: seems all the compression class definition can be moved into cpp file.

File: be/src/util/compression/stream_compression.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <bzlib.h>
+#include <lz4/lz4frame.h>
+#include <zlib.h>
+#include <zstd/zstd.h>
+#include <zstd/zstd_errors.h>
```
Comment (imay) [inline be/src/util/compression/stream_compression.h:28]: These includings can be also moved into cpp

File: be/src/util/compression/stream_compression.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <memory>
+
+#include "common/status.h"
+#include "gen_cpp/types.pb.h"
+#include "gutil/strings/substitute.h"
+#include "util/compression/compression_context_pool_singletons.h"
```
Comment (imay) [inline be/src/util/compression/stream_compression.h:35]: is this including neccessary?

Comment (xiaoyong-z) [reply be/src/util/compression/stream_compression.h]: some of them are necessary

**Review by imay — APPROVED** [review]

**Pull request merged**